hail_search/queries/both_sample_types_mixin.py

```diff
@@ -0,0 +1,141 @@
+            wgs_ht = wgs_ht.rename({'family_entries': 'wgs_family_entries', 'filters': 'wgs_filters'})
+            ht = wes_ht.join(wgs_ht, how='outer')
+
+            ht = self._filter_quality_both_sample_types(ht, quality_filter)
+            ht, ch_ht = self._filter_inheritance_both_sample_types(
+                ht, inheritance_filter, sorted_wes_family_sample_data, sorted_wgs_family_sample_data
+            )
+            if ht:
+                filtered_project_hts.append(self._merge_both_sample_types_project_ht(ht))
+            if ch_ht:
+                filtered_comp_het_project_hts.append(self._merge_both_sample_types_project_ht(ch_ht))
+
+        return filtered_project_hts, filtered_comp_het_project_hts
+
+    def _filter_quality_both_sample_types(self, ht: hl.Table, quality_filter: dict):
+        """
+        Filter by quality, keeping variants that pass quality filter in at least one sample type.
+        """
+        wes_passes_quality_filter = self._get_family_passes_quality_filter(quality_filter, ht, 'wes_filters')
+        if wes_passes_quality_filter is not None:
+            ht = ht.annotate(
+                wes_passes=ht.wes_family_entries.map(
+                    lambda entries: hl.or_missing(wes_passes_quality_filter(entries), entries)
+                ))
+        else:
+            ht = ht.annotate(wes_passes=ht.wes_family_entries)
+        wgs_passes_quality_filter = self._get_family_passes_quality_filter(quality_filter, ht, 'wgs_filters')
+        if wgs_passes_quality_filter is not None:
+            ht = ht.annotate(
+                wgs_passes=ht.wgs_family_entries.map(
+                    lambda entries: hl.or_missing(wgs_passes_quality_filter(entries), entries)
+                ))
+        else:
+            ht = ht.annotate(wgs_passes=ht.wgs_family_entries)
+        ht = ht.filter(ht.wes_passes.any(hl.is_defined) | ht.wgs_passes.any(hl.is_defined))
+        return ht
+
+    @staticmethod
+    def _merge_both_sample_types_project_ht(ht: hl.Table):
+        ht = ht.drop('wes_passes', 'wgs_passes')
+        ht = ht.transmute(
+            family_entries=hl.coalesce(ht.wes_family_entries, hl.empty_array(ht.wes_family_entries[0].dtype)).extend(
+                hl.coalesce(ht.wgs_family_entries, hl.empty_array(ht.wgs_family_entries[0].dtype))
+            ),
+            filters=hl.coalesce(ht.wes_filters, hl.empty_set(hl.tstr)).union(
+                hl.coalesce(ht.wgs_filters, hl.empty_set(hl.tstr))
+            )
+        )
+        return ht.transmute_globals(family_guids=ht.family_guids.extend(ht.family_guids_1))
+
+
+    def _filter_inheritance_both_sample_types(
+        self, ht, inheritance_filter, sorted_wes_family_sample_data, sorted_wgs_family_sample_data
+    ):
+        # At least one family member must have non-ref gt
+        any_valid_entry, is_any_affected = self._get_any_family_member_gt_has_alt_filter()
+        ht = ht.annotate(
+            wes_passes=ht.wes_family_entries.map(
```

Comment:
shouldn't this be `ht.wes_passes.map` so we avoid failed quality families

Reply:
I think we want to check quality and inheritance independent of each other passing or failing in this case